File: src/anylayout/util/MapUtility.java
package anylayout.util;

import fj.F;

import java.util.Map;

public class MapUtility
{
	public static <K,V> V get(final Map<K,V> map,final K key,final F<K,V> defaultValue)
	{
		final V value=map.get(key);

		return null==value ? defaultValue.f(key) : value;
	}
}

File: src/anylayout/util/MapUtility.scala
package anylayout.util

import java.util.Map

object MapUtility {
 def get[K, V](map: Map[K, V], key: K, defaultValue: K => V) = { val value = map.get(key)
                                                                 if (null==value) defaultValue(key) else value }
}
